libs/core/hardware/include/hpx/hardware/timestamp.hpp

```diff
@@ -11,7 +11,11 @@
 #include <hpx/config.hpp>
 
 // clang-format off
-#if defined(HPX_MSVC)
+// nvcc complains about multiple definition error for same function prototype
+// irrespective of __host__, __device__, and __host__ __device__ attributes. 
```

Comment:
Could you explain what the problem is here?

Reply:
I just realized that this implementation won't work either. Let me clarify what I meant:
clang and nvcc views same function prototypes with different attributes (__host__, __device__ or __host__ __device__) differently. For clang, `__host__ void foo()` and `__device__ void foo()` are 2 different function prototypes. But nvcc views them as the same function despite having 2 completely different attributes. nvcc in these cases throws multiple definition errors and the `#if defined(__CUDA_ARCH__)` is used to achieve a similar scenario as clang. So, with nvcc, you would defined one function foo and use the macro to check whether the current execution path is on GPU. I used this technique in the previous commit where I define the function HPX_HOST_DEVICE and use the macro to define 2 separate implementation paths depending on GPU/CPU execution. I realized that the macro won't help here as it's not used within a function.
I'll have to come up with something that integrates well with the current system and doesn't lead to multiple declaration error with nvcc. I'll update the PR again with that.